numba/core/interpreter.py

```diff
@@ -2859,23 +2859,36 @@ def op_JUMP_IF_TRUE(self, inst, pred):
         self._op_JUMP_IF(inst, pred=pred, iftrue=True)
 
     def _jump_if_none(self, inst, pred, iftrue):
-        brs = {
-            True: inst.get_jump_target(),
-            False: inst.next,
-        }
-        truebr = brs[iftrue]
-        falsebr = brs[not iftrue]
+        # branch pruning assumes true falls through and false is jump
+        truebr = inst.next
+        falsebr = inst.get_jump_target()
+
+        # this seems strange
+        if not iftrue:
+            op = BINOPS_TO_OPERATORS["is"]
+        else:
+            op = BINOPS_TO_OPERATORS["is not"]
```

Comment:
`inst.next` is the fallthrough branch. `inst.get_jump_target()` is the take-the-jump branch. need to fix up the `truebr` `falsebr` variable naming.